sentry-ruby/lib/sentry/net/http.rb

```diff
@@ -26,6 +55,13 @@ def do_finish
 
       private
 
+      def set_sentry_trace_header(req)
+        return unless @sentry_span
+
+        trace = Sentry.get_current_hub.generate_sentry_trace(@sentry_span)
```

Comment:
@rhcarvalho `Hub#generate_sentry_trace` calls performs `propagate_traces` config check, logs the message and then call `Span#to_sentry_trace`. [sentry-python](https://github.com/getsentry/sentry-python/blob/51987c57157102bbd32e1e7b084c26f4dc475d86/sentry_sdk/integrations/stdlib.py#L88) also takes the similar approach.

Reply:
I missed that, sorry. I wanted to minimize the amount of SDK-specific names and ad-hoc APIs, but that's tough 😅 

Any chance we could inline the `generate_sentry_trace` code here to avoid having one more method on `Hub`?

Reply:
I can move it to `Client` instead if you think that's better? I want to keep most of the logic outside the `HTTP` patch so they can be easier for testing.

Reply:
If testing is the concern, then how about this being a function

```ruby
def generate_sentry_trace(client, span)
 ...
end
```

This way we don't expose any new methods on standard parts of the SDK API (Hub, Client, Scope).
I'd say we don't need the `Hub` because it is only used to get the current scope when the span is `nil`, but we never call it with a `nil` span.
The `Client` is used to look up the config and to access the logger.

If we make it a function with all explicit inputs, then we can test it easy by passing whatever client/span we want, even no need to monkey-patch anything.

What do you think?

Reply:
where should we put this method? if we define it at the top level it'll be visible to the entire Ruby program, which is not ideal. so eventually we'll still need to find a place for it. I don't think placing it in the `Net::HTTP` (stdlib) is better than having it in `Hub`.

Reply:
another (but not the main) consideration is that, some Ruby applications use other http libraries still won't have traces propagation. so if we want to support them in the future, this  method needs to be outside of the http lib.

Reply:
Yeah definitely not in the stdlib.
Don't we have any utils module?

Reply:
If nowhere else, then in the Client since this is more about applying/checking some client configuration before calling a span method.